server/app/services/geo/esri/EsriServiceAreaValidationConfig.java

```diff
@@ -0,0 +1,124 @@
+        || this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS.isEmpty()
+        || this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_PATHS.isEmpty()) {
+      return false;
+    }
+
+    return true;
+  }
+
+  private Optional<ImmutableList<String>> getConfigListSetting(Optional<ConfigList> setting) {
+    if (setting.isEmpty()) {
+      return Optional.empty();
+    }
+
+    ImmutableList.Builder<String> listBuilder = ImmutableList.builder();
+    setting.get().stream()
+        .forEach(configValue -> listBuilder.add((String) configValue.unwrapped()));
+    return Optional.of(listBuilder.build());
+  }
+
+  /** Returns the labels specified for Esri address service area validation. */
+  public Optional<ImmutableList<String>> getLabels() {
+    return getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS);
+  }
+
+  /** Returns the values specified for Esri address service area validation. */
+  public Optional<ImmutableList<String>> getValues() {
+    return getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_VALUES);
+  }
+
+  /** Returns the urls specified for Esri address service area validation. */
+  public Optional<ImmutableList<String>> getUrls() {
+    return getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS);
+  }
+
+  /** Returns the paths specified for Esri address service area validation. */
+  public Optional<ImmutableList<String>> getPaths() {
+    return getConfigListSetting(this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_PATHS);
+  }
+
+  /**
+   * Transforms the config settings for Esri address service area validation into an optional
+   * immutable list of {@link EsriServiceAreaValidationOption}s.
+   */
+  private Optional<ImmutableList<EsriServiceAreaValidationOption>> toImmutableList() {
+    if (!hasAllElements()) {
+      return Optional.empty();
+    }
+
+    ImmutableList<String> labels = getLabels().get();
+    ImmutableList<String> values = getValues().get();
+    ImmutableList<String> urls = getUrls().get();
+    ImmutableList<String> paths = getPaths().get();
+
+    ImmutableList.Builder<EsriServiceAreaValidationOption> listBuilder = ImmutableList.builder();
+
+    for (int i = 0; i < values.size(); i++) {
+      EsriServiceAreaValidationOption option =
+          EsriServiceAreaValidationOption.builder()
+              .setLabel(labels.get(i))
+              .setValue(values.get(i))
+              .setUrl(urls.get(i))
+              .setPath(paths.get(i))
+              .build();
+
+      listBuilder.add(option);
+    }
+
+    return Optional.of(listBuilder.build());
+  }
+
+  /**
+   * Returns an {@link EsriServiceAreaValidationOption} from the config list for Esri address
+   * service area settings given a service area.
+   */
+  public Optional<EsriServiceAreaValidationOption> getOptionByServiceArea(String serviceArea) {
+    Optional<ImmutableList<EsriServiceAreaValidationOption>> list = toImmutableList();
+
+    if (list.isEmpty()) {
+      return Optional.empty();
+    }
```

Comment:
**suggestion (blocking):** Add error logging anywhere we're returning an empty optional because of an error case

Reply:
added logging